File: app/src/main/java/com/blackmirror/dongda/ui/activity/ListMoreContract.java
package com.blackmirror.dongda.ui.activity;

import com.blackmirror.dongda.domain.model.BaseDataBean;
import com.blackmirror.dongda.domain.model.CareMoreDomainBean;
import com.blackmirror.dongda.domain.model.LikePopDomainBean;
import com.blackmirror.dongda.domain.model.LikePushDomainBean;
import com.blackmirror.dongda.ui.BasePresenter;

/**
 * Create By Ruge at 2018-05-11
 */
public interface ListMoreContract {

    interface View{

        void onGetServiceMoreDataSuccess(CareMoreDomainBean bean);

        void onLikePushSuccess(LikePushDomainBean bean);

        void onLikePopSuccess(LikePopDomainBean bean);

        void onGetDataError(BaseDataBean bean);

    }
    interface presenter extends BasePresenter<View>{

        void getServiceMoreData(int skipCount, int takeCount, String serviceType);

        void likePush(String service_id);

        void likePop(String service_id);
    }
}

File: app/src/main/java/com/blackmirror/dongda/ui/activity/ListMoreContract.kt
package com.blackmirror.dongda.ui.activity

import com.blackmirror.dongda.kdomain.model.BaseDataBean
import com.blackmirror.dongda.kdomain.model.CareMoreDomainBean
import com.blackmirror.dongda.kdomain.model.LikePopDomainBean
import com.blackmirror.dongda.kdomain.model.LikePushDomainBean
import com.blackmirror.dongda.ui.BasePresenter

/**
 * Create By Ruge at 2018-05-11
 */
interface ListMoreContract {

    interface View {

        fun onGetServiceMoreDataSuccess(bean: CareMoreDomainBean)

        fun onLikePushSuccess(bean: LikePushDomainBean)

        fun onLikePopSuccess(bean: LikePopDomainBean)

        fun onGetDataError(bean: BaseDataBean)

    }

    interface Presenter : BasePresenter<View> {

        fun getServiceMoreData(skipCount: Int, takeCount: Int, serviceType: String)

        fun likePush(service_id: String)

        fun likePop(service_id: String)
    }
}
